Tests/Common/SimulatedDevice.cs

```diff
@@ -265,73 +264,90 @@ private bool HandleJoinAccept(LoRaPayloadJoinAccept payload)
-                await basicsStation.SerializeAndSendMessageAsync(new
+                try
                 {
-                    JoinEui = joinRequestPayload.AppEui.ToString("D", null),
-                    msgtype = "jreq",
-                    DevEui = joinRequestPayload.DevEUI.ToString("D", null),
-                    DevNonce = joinRequestPayload.DevNonce.AsUInt16,
-                    MHdr = uint.Parse(joinRequestPayload.MHdr.ToString(), NumberStyles.HexNumber, CultureInfo.InvariantCulture),
-                    MIC = joinRequestPayload.Mic.Value.AsInt32,
-                    DR = joinRequest.RadioMetadata.DataRate,
-                    Freq = joinRequest.RadioMetadata.Frequency.AsUInt64,
-                    upinfo = new
+                    foreach (var basicsStation in this.simulatedBasicsStations)
                     {
-                        gpstime = joinRequest.RadioMetadata.UpInfo.GpsTime,
-                        rctx = 10,
-                        rssi = joinRequest.RadioMetadata.UpInfo.ReceivedSignalStrengthIndication,
-                        xtime = joinRequest.RadioMetadata.UpInfo.Xtime,
-                        snr = joinRequest.RadioMetadata.UpInfo.SignalNoiseRatio
+                        await basicsStation.SerializeAndSendMessageAsync(new
+                        {
+                            JoinEui = joinRequestPayload.AppEui.ToString("D", null),
+                            msgtype = "jreq",
+                            DevEui = joinRequestPayload.DevEUI.ToString("D", null),
+                            DevNonce = joinRequestPayload.DevNonce.AsUInt16,
+                            MHdr = uint.Parse(joinRequestPayload.MHdr.ToString(), NumberStyles.HexNumber,
+                                              CultureInfo.InvariantCulture),
+                            MIC = joinRequestPayload.Mic.Value.AsInt32,
+                            DR = joinRequest.RadioMetadata.DataRate,
+                            Freq = joinRequest.RadioMetadata.Frequency.AsUInt64,
+                            upinfo = new
+                            {
+                                gpstime = joinRequest.RadioMetadata.UpInfo.GpsTime,
+                                rctx = 10,
+                                rssi = joinRequest.RadioMetadata.UpInfo.ReceivedSignalStrengthIndication,
+                                xtime = joinRequest.RadioMetadata.UpInfo.Xtime,
+                                snr = joinRequest.RadioMetadata.UpInfo.SignalNoiseRatio
+                            }
+                        });
                     }
-                });
-            }
 
-#if DEBUG
-            if (System.Diagnostics.Debugger.IsAttached)
-            {
-                timeout = TimeSpan.FromSeconds(60);
+                    return await tcs.Task.WaitAsync(timeout);
+                }
+                catch (Exception ex)
+                {
+                    this.logger.LogError(ex, "Join attempt failed due to: '{Message}'.", ex.Message);
```

Comment:
Should we be treating any and every error as a failure to join?

Reply:
I think we should attempt to retry always, even if the LNS does not send a response (timeout), this makes the tests more stable. It's only used for the load tests, and we log all the errors.